src/Util.js

```diff
@@ -64,10 +64,35 @@ class ErrorContext extends Error {
     }
 }
 
+// Represents a PR 'identificator' as a string, either of
+// PR number, or
+// PR branch name (without 'refs' or 'heads' prefixes), or
+// staging branch commit SHA (including stale commits).
+class PrId
+{
+    constructor(type, val, msg) {
+        assert(type !== undefined);
+        assert(type !== null);
+        assert(val !== undefined);
+        assert(val !== null);
+        this.type = type; // a PR identificator type ("branch", "sha" or "prNum")
+        this.value = val; // a PR identificator
+        this.message = (msg === undefined) ? null : msg; // the commit message or null
+    }
+
+    static BranchList(branches, msg) { return Array.from(branches, b => new PrId("branch", b, msg)); }
+    static Sha(sha) { return [new PrId("sha", sha)]; }
+    static PrNum(prNum) { return [new PrId("prNum", prNum.toString())]; }
+    static PrNumList(list) { return Array.from(list, prNum => new PrId("prNum", prNum.toString())); }
```

Comment:
Why do we need to convert PR numbers to strings? If it is not going to cause problems, let's store them as integers/numbers. There are other similarly strange expressions like `id.value.toString()` elsewhere that will probably go away if we do not have to convert.